MapleServer2/PacketHandlers/Game/EnchantScrollHandler.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Maple2Storage.Enums;
+using Maple2Storage.Types.Metadata;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Data.Static;
+using MapleServer2.PacketHandlers.Game.Helpers;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using MapleServer2.Tools;
+using MapleServer2.Types;
+
+namespace MapleServer2.PacketHandlers.Game;
+
+public class EnchantScrollHandler : GamePacketHandler
+{
+    public override RecvOp OpCode => RecvOp.EnchantScroll;
+
+    private enum EnchantScrollMode : byte
+    {
+        AddItem = 0x1,
+        UseScroll = 0x2,
+    }
+
+    private enum EnchantScrollError : short
+    {
+        None = 0x0,
+        ItemsNoLongerValid = 0x1,
+        IneligibleItem = 0x2,
+        UnstableItemCannotBeEnchanted = 0x3,
+        InsufficientItemLevel = 0x4,
+        GearCannotBeEnchanted = 0x5,
+        CannotBeEnchantedDueToRarity = 0x6,
+        GearRarityIsHigher = 0x7,
+        UnstableItemCannotBeRestored = 0x8,
+    }
+
+    public override void Handle(GameSession session, PacketReader packet)
+    {
+        EnchantScrollMode mode = (EnchantScrollMode) packet.ReadByte();
+
+        long scrollUid = packet.ReadLong();
+        long equipUid = packet.ReadLong();
+
+        if (!session.Player.Inventory.HasItem(equipUid) && !session.Player.Inventory.HasItem(scrollUid))
+        {
+            session.Send(EnchantScrollPacket.UseScroll((short) EnchantScrollError.ItemsNoLongerValid));
+            return;
+        }
+
+        Item scroll = session.Player.Inventory.GetByUid(scrollUid);
+        Item equip = session.Player.Inventory.GetByUid(equipUid);
+
+        EnchantScrollMetadata metadata = EnchantScrollMetadataStorage.GetMetadata(scroll.Function.Id);
+        if (metadata is null)
+        {
+            return;
+        }
+
+        if (!metadata.Rarities.Contains(equip.Rarity))
+        {
+            session.Send(EnchantScrollPacket.UseScroll((short) EnchantScrollError.CannotBeEnchantedDueToRarity));
+            return;
+        }
+
+        if (metadata.MinLevel > equip.Level || metadata.MaxLevel < equip.Level)
+        {
+            session.Send(EnchantScrollPacket.UseScroll((short) EnchantScrollError.InsufficientItemLevel));
+            return;
+        }
+
+        if (!metadata.ItemTypes.Contains(equip.Type))
+        {
+            session.Send(EnchantScrollPacket.UseScroll((short) EnchantScrollError.IneligibleItem));
+            return;
+        }
+
+        if (equip.EnchantLevel >= metadata.EnchantLevels.Last())
+        {
+            session.Send(EnchantScrollPacket.UseScroll((short) EnchantScrollError.GearCannotBeEnchanted));
+            return;
+        }
+
+        Random random = Random.Shared;
+
+        int enchantLevelIndex = random.Next(metadata.EnchantLevels.Count);
```

Comment:
```suggestion
        int enchantLevelIndex = Random.Shared.Next(metadata.EnchantLevels.Count);
```

No reason to pass Random to the function HandleUseScroll. Just call Random.Shared inside